apps/daimo-web/src/middleware.ts

```diff
@@ -0,0 +1,26 @@
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+
+export function middleware(request: NextRequest) {
+  if (request.nextUrl.basePath === "/l/download") {
+    return redirectMobileToAppStore(request);
+  }
+  return NextResponse.next();
+}
+
+function redirectMobileToAppStore(request: NextRequest) {
+  const ua = request.headers.get("user-agent") || "";
+  const url = getAppStoreURL(ua);
+  if (url) return NextResponse.redirect(url);
+  return NextResponse.next();
+}
+
+function getAppStoreURL(ua: string) {
+  const urlAppStore = "https://apps.apple.com/us/app/daimo/id6459700343";
+  const urlPlay = "https://play.google.com/store/apps/details?id=com.daimo";
+  if (ua.includes("Android")) return urlPlay;
+  else if (ua.includes("iPhone")) return urlAppStore;
```

Comment:
looks like we have some utils for this same heuristic in https://github.com/daimo-eth/daimo/blob/master/apps/daimo-web/src/utils/platform.ts, want to dedupe?

Reply:
good call, done